lib/babel/transformation/transformers/es6/parameters.rest.js

```diff
@@ -19,6 +77,24 @@ exports.Function = function (node, parent, scope) {
   // otherwise `arguments` will be remapped in arrow functions
   argsId._ignoreAliasFunctions = true;
 
+  // check if rest is used only in member expressions
+  var restOuterDeclar = scope.getBindingIdentifier(rest.name);
+  var state = {
+    name: rest.name,
+    outerDeclar: restOuterDeclar,
+    isOptimizable: true,
+    candidates: []
+  };
+  scope.traverse(node, memberExpressionVisitor, state);
+
+  if (state.isOptimizable) {
```

Comment:
This is not about access to member expression, but about this:

    function m(foo) {
      for (var _len = arguments.length, bar = Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
        bar[_key - 1] = arguments[_key];
      }

      var x = arguments[1 + 1 + 1];
      var y = arguments[x + 5 + 1];
      var z = bar[x || y];
    }

It would replace some references without knowing if deopt is needed. It would need second pass to repair this (or it could be repaired by using the same candidate list as before - no gain).